sky/skylet/providers/aws/config.py

```diff
@@ -325,12 +345,42 @@ def _configure_iam_role(config):
 
             for policy_arn in attach_policy_arns:
                 role.attach_policy(PolicyArn=policy_arn)
+                        
+            # SkyPilot: "PassRole" is required by the head node to pass the role to
+            # the workers, so we can access S3 buckets on the workers. "Resource"
+            # is to limit the role to only able to pass itself to the workers.
+            skypilot_pass_role_policy_doc = {
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "iam:GetRole",
+                            "iam:PassRole",
+                        ],
+                        "Resource": role.arn,
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": "iam:GetInstanceProfile",
+                        "Resource": profile.arn,
+                    }
+                ]
+            }
+            role.Policy("SkypilotPassRolePolicy").put(
```

Comment:
Will this possibly assign the policy to the Ray role? I see that we will always use V2 for new clusters; however, should we guard with `if skypilot_iam_role:` to be safe?

Reply:
Good point! Added. Thanks!